src/Async.js

```diff
@@ -77,6 +77,10 @@ class Async extends React.Component {
       case 1:
         return this.props.children(value);
       case 2:
+        if (this.props.catch) {
+          return this.props.catch(error);
```

Comment:
could we do this instead?

```js
if (this.props.catch) {
  return this.props.children(this.props.catch(error));
}
```

that way all of the render logic can be in one place

```js
<Async
  await={promise}
  catch={error => ({ error })}
>
  {result => (
    <div>
      {result.error && <h1>{result.error.message}</h1>}
    </div>
  )}
</Async>
```